mozilla/foundation.mozilla.org
# updated JS to toggle category CTA

# Description

Link to sample test page: http://localhost:8000/en/privacynotincluded/categories/entertainment/
Related PRs/issues: #9853 

This PR adds a new util function, titled `toggleCtaForCategory`, and adds it to the function `filterCategory`, which is called whenever the user selects a category from either the category/subcategory navigation bars.

`toggleCtaForCategory` then checks whether or not the selected category can be found in the featured CTA's `data-show-for-categories` attribute, and renders the featured CTA on the category page accordingly. 


## Steps to test:

1. Pull this branch and run `inv new-db`
2. Visit a category & subcategory of your choice in the CMS, and select the `show_cta` checkbox.
3. Load the category page (EX: privacynotincluded/categories/entertainment) for the category with `show_cta = 
True`, you should see the featured CTA on page render.
4. Load the category page for a category with `show_cta = False` you **should not** see the CTA on page render.
5. Load the category page for a category with `show_cta = False` you **should not see** the featured CTA, but when you switch the to a category/subcategory with `show_cta = True`, the featured CTA should appear on the page. Selecting a category with is set to `False` should also remove the featured CTA from the page.
6. If everything is working as expected, testing is complete!
